spec/services/catalog/create_approval_request_spec.rb

```diff
@@ -0,0 +1,86 @@
+describe Catalog::CreateApprovalRequest do
+  let(:create_approval_request) { described_class.new(order.id) }
+
+  let(:workflow_ref) { "1" }
+  let!(:order) { create(:order) }
+  let!(:portfolio) { create(:portfolio) }
+  let!(:portfolio_item) { create(:portfolio_item, :portfolio_id => portfolio.id, :workflow_ref => workflow_ref) }
+  let!(:order_item) { create(:order_item, :order_id => order.id, :portfolio_item_id => portfolio_item.id) }
+
+  let(:approval) { class_double(Approval).as_stubbed_const(:transfer_nested_constants => true) }
+  let(:api_instance) { instance_double(ApprovalApiClient::RequestApi) }
+  let(:approval_response) { ApprovalApiClient::RequestOut.new(:id => 1, :decision => "undecided", :workflow_id => workflow_ref) }
+
+  let(:sanitize_service_class) do
+    class_double(Catalog::OrderItemSanitizedParameters).as_stubbed_const(:transfer_nested_constants => true)
+  end
+  let(:sanitize_service_instance) { instance_double(Catalog::OrderItemSanitizedParameters) }
+  let(:hashy) { { :a => 1 } }
+
+  before do
+    allow(sanitize_service_class).to receive(:new).and_return(sanitize_service_instance)
+    allow(sanitize_service_instance).to receive(:process).and_return(hashy)
+    allow(approval).to receive(:call).and_yield(api_instance)
+    allow(api_instance).to receive(:create_request).and_return(approval_response)
+  end
+
+  context "#process" do
```

Comment:
I believe `describe` makes more sense for specific methods http://www.betterspecs.org/

Reply:
Bookmarking that site!